Index: kgraphviewer_part.cpp
===================================================================
--- kgraphviewer_part.cpp	(révision 152)
+++ kgraphviewer_part.cpp	(copie de travail)
@@ -19,7 +19,7 @@
 
 #include "kgraphviewer_part.h"
 
-#include <kinstance.h>
+#include <kcomponentdata.h>
 #include <kaction.h>
 #include <ktoggleaction.h>
 #include <kselectaction.h>
@@ -50,7 +50,7 @@
 {
 //   std::cerr << "kgraphviewerPart::kgraphviewerPart" << std::endl;
   // we need an instance
-  setInstance( kgraphviewerPartFactory::instance() );
+  setComponentData( kgraphviewerPartFactory::componentData() );
 
   // this should be your custom internal widget
   m_widget = new DotGraphView( actionCollection(), parentWidget);
@@ -137,7 +137,7 @@
   }
 }
 
-KInstance*  kgraphviewerPartFactory::s_instance = 0L;
+KComponentData kgraphviewerPartFactory::s_instance = 0L;
 KAboutData* kgraphviewerPartFactory::s_about = 0L;
 
 kgraphviewerPartFactory::kgraphviewerPartFactory()
@@ -168,7 +168,7 @@
     return obj;
 }
 
-KInstance* kgraphviewerPartFactory::instance()
+KComponentData kgraphviewerPartFactory::componentData()
 {
     if( !s_instance )
     {
@@ -176,7 +176,7 @@
                     "1.0", I18N_NOOP( "GraphViz dot files viewer" ),
                     KAboutData::License_GPL,
                     "(c) 2005-2006, Gaël de Chalendar <kleag@free.fr>");
-        s_instance = new KInstance(s_about);
+        s_instance(s_about);
     }
     return s_instance;
 }
Index: kgraphviewer_part.h
===================================================================
--- kgraphviewer_part.h	(révision 152)
+++ kgraphviewer_part.h	(copie de travail)
@@ -70,7 +70,7 @@
   KDirWatch* m_watch;
 };
 
-class KInstance;
+class KComponentData;
 class KAboutData;
 
 class kgraphviewerPartFactory : public KParts::Factory
@@ -82,10 +82,10 @@
     virtual KParts::Part* createPartObject( QWidget *parentWidget, 
                                             QObject *parent, 
                                             const char *classname, const QStringList &args );
-    static KInstance* instance();
+    static KComponentData componentData();
  
 private:
-    static KInstance* s_instance;
+    static KComponentData s_instance;
     static KAboutData* s_about;
 };
